kloppy/infra/serializers/event/metrica/json_serializer.py

```diff
@@ -0,0 +1,311 @@
+                    dictionary.
+                Returns
+                -------
+                dataset : EventDataset
+                Raises
+                ------
+
+                See Also
+                --------
+
+                Examples
+                --------
+                >>> serializer = MetricaEventsJsonSerializer()
+                >>> with open("events.json", "rb") as event_data, \
+                >>>      open("metadata.xml", "rb") as metadata:
+                >>>
+                >>>     dataset = serializer.deserialize(
+                >>>         inputs={
+                >>>             'event_data': event_data,
+                >>>             'metadata': metadata
+                >>>         },
+                >>>         options={
+                >>>             'event_types': ["pass", "take_on", "carry", "shot"]
+                >>>         }
+                >>>     )
+                """
+        self.__validate_inputs(inputs)
+        if not options:
+            options = {}
+
+        with performance_logging("load data", logger=logger):
+            raw_events = json.load(inputs["event_data"])
+            metadata = load_metadata(
+                inputs["metadata"], provider=Provider.METRICA
+            )
+
+        with performance_logging("parse data", logger=logger):
+
+            wanted_event_types = [
+                EventType[event_type.upper()]
+                for event_type in options.get("event_types", [])
+            ]
+
+            events = []
+            for raw_event in raw_events["data"]:
+                if raw_event["team"]["id"] == metadata.teams[0].team_id:
+                    team = metadata.teams[0]
+                elif raw_event["team"]["id"] == metadata.teams[1].team_id:
+                    team = metadata.teams[1]
+                else:
+                    raise Exception(
+                        f"Unknown team_id {raw_event['team']['id']}"
+                    )
+
+                player = team.get_player_by_id(raw_event["from"]["id"])
+                event_type = raw_event["type"]["id"]
+                subtypes = _parse_subtypes(raw_event)
+                period = [
+                    period
+                    for period in metadata.periods
+                    if period.id == raw_event["period"]
+                ][0]
+
+                generic_event_kwargs = dict(
+                    # from DataRecord
+                    period=period,
+                    timestamp=raw_event["start"]["time"],
+                    ball_owning_team=_parse_ball_owning_team(event_type, team),
+                    ball_state=BallState.ALIVE,
+                    # from Event
+                    event_id=None,
+                    team=team,
+                    player=player,
+                    coordinates=(_parse_coordinates(raw_event["start"])),
+                    raw_event=raw_event,
+                )
+
+                if event_type in MS_PASS_TYPES:
```

Comment:
Could consider adding a type check here so you can skip parsing completely if the event type was not listed in the options.
```
event = None
if event_type in MS_PASS_TYPES:
    if EventType.PASS not in wanted_event_types:
        continue
...
```
line 232 would need to change to:
`for event_type in options.get("event_types", [e for e in EventType])`
and line 302 changed to:
```
if event:
    events.append(event)
```

Reply:
Personally I like to have a less as branches (if ... ) as possible for readability. In case when performance really suffers it might be good do skip parsing. That's not the case here.

Furthermore keeping the same structure for each serializer allows us to do a refactor and extract common parts to a generic event serializer / base event serializer.